### PR TITLE
Add validating webhook for VerrazzanoProject resource

### PR DESCRIPTION
**Pull request by mgianatagh**
# Description

Add a validating webhook for the VerrazzanoProject resource.  Add unit and integration tests too. 

Fixes VZ-2212

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/apis/clusters/v1alpha1/verrazzanoproject_webhook.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var _ webhook.Validator = &VerrazzanoProject{}
+
+// log is for logging in this package.
+var log = logf.Log.WithName("ingresstrait-resource")
```
Comment (kminder) [inline application-operator/apis/clusters/v1alpha1/verrazzanoproject_webhook.go:22]: Log name shouldn't be "ingresstrait-resource", right?

**Comment by kminder** [comment]
Can you please update https://github.com/verrazzano/verrazzano/blob/master/application-operator/Makefile#L58 with the commands required to disable this new webhook.

File: application-operator/apis/clusters/v1alpha1/verrazzanoproject_webhook.go (hunk 1 above)
Comment (mgianatagh) [reply application-operator/apis/clusters/v1alpha1/verrazzanoproject_webhook.go:22]: oops, not it shouldn't

Comment (mgianatagh) [reply application-operator/apis/clusters/v1alpha1/verrazzanoproject_webhook.go]: fixed

**Comment by mgianatagh** [comment]
> Can you please update https://github.com/verrazzano/verrazzano/blob/master/application-operator/Makefile#L58 with the commands required to disable this new webhook.

I used the same flag for enabling the existing webhooks.  Should I create a separate flag to control enable/disable of this new webhook?

File: application-operator/deploy/verrazzano.yaml_template [hunk 2]
```diff
@@ -167,3 +167,30 @@ webhooks:
           - pods
     sideEffects: None
     failurePolicy: Fail
+---
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: verrazzano-application-verrazzanoproject-validator
```
Comment (pfmackin) [inline application-operator/deploy/verrazzano.yaml_template:174]: using -validator is redundant, the type is ValidatingWebhookConfiguration

**Review by pfmackin — APPROVED** [review]

**Review by desagar — APPROVED** [review]

**Pull request merged**